### ISSUE TITLE
Feature request: cache in OS tmpdir

### ISSUE THREAD

**Issue by tbranyen** [mapbox/node-pre-gyp#149]
It would be great if node-pre-gyp could cache the fetched bundle from Amazon S3 and simply default to that first if it matches version and whatever criteria node-pre-gyp uses to determine the filename.

Here's a reference issue in NodeGit where we noticed this: https://github.com/nodegit/nodegit/issues/379

**Comment by springmeyer** [comment]
Yes, this would be a nice feature. But it would need good tests to ensure that any corrupt or partially downloaded binaries would not cause headaches. Would you be interested in providing a pull request? See also one attempt at this over at #101 that never landed because a pull+tests was never provided.
